### PR TITLE
Add two new time profiling events

### PR DESCRIPTION
**Pull request by mpickering**
The eventlog now contains two new events which are emitted when `-p` is
enabled.

1. ProfSampleCostCentre - The current cost centre stack on each tick
2. ProfBegin - Start of time profile to report the profiling interval

https://gitlab.haskell.org/ghc/ghc/merge_requests/1927

### CODE REVIEW

File: ghc-events.cabal [hunk 1]
```diff
@@ -61,6 +61,8 @@ extra-source-files: include/EventLogFormat.h
                     test/hello-ghc-8.6.5.eventlog.reference
                     test/biographical-samples.eventlog
                     test/biographical-samples.eventlog.reference
+                    test/time-prof.eventlog
+                    test/time-prog.eventlog.reference
```
Comment (maoe) [inline ghc-events.cabal]: s/prog/prof/. I think this is the culprit of the CI failures.

File: include/EventLogFormat.h [hunk 2]
```diff
@@ -197,13 +197,15 @@
...
+#define EVENT_PROF_SAMPLE_COST_CENTRE      167
+#define EVENT_PROF_BEGIN                   168
 
 /*
  * The highest event code +1 that ghc itself emits. Note that some event
  * ranges higher than this are reserved but not currently emitted by ghc.
  * This must match the size of the EventDesc[] array in EventLog.c
  */
-#define NUM_GHC_EVENT_TAGS        167
+#define NUM_GHC_EVENT_TAGS        181
```
Comment (maoe) [inline include/EventLogFormat.h]: I think I should take a look at the GHC code but were that many (~12) events added/reserved to/in the eventlog framework recently?

File: src/GHC/RTS/EventTypes.hs [hunk 3]
```diff
@@ -418,6 +418,16 @@ data EventInfo
                        , heapProfResidency :: !Word64
                        , heapProfLabel :: !Text
                        }
+
+  | ProfSampleCostCentre
+                       { profCapset :: Capset
```
Comment (maoe) [inline src/GHC/RTS/EventTypes.hs]: Is there a reason not to make this strict?

File: include/EventLogFormat.h (hunk 2 above)
Comment (maoe) [reply include/EventLogFormat.h]: I see now. 160-180 is reserved for heap profiling events.

File: src/GHC/RTS/EventTypes.hs (hunk 3 above)
Comment (mpickering) [reply src/GHC/RTS/EventTypes.hs]: I made it strict now.

File: ghc-events.cabal (hunk 1 above)
Comment (mpickering) [reply ghc-events.cabal]: Thanks.

File: src/GHC/RTS/EventTypes.hs (hunk 3 above)
Comment (maoe) [reply src/GHC/RTS/EventTypes.hs]: Thanks!

**Comment by maoe** [comment]
LGTM. Thank you!

**Comment by maoe** [comment]
Released https://hackage.haskell.org/package/ghc-events-0.11.0

**Pull request merged**